product_listings_manager/products.py

```diff
@@ -81,122 +77,65 @@ def execute_query(dbh, query, **kwargs):
+    def get_overrides(product, version, variant=None):
         '''Returns the list of package overrides for the particular product specified.'''
 
-        qargs = dict(product=product, version=version)
+        query = models.Overrides.query.join(models.Overrides.productref).filter(
+            models.Products.label == product, models.Products.version == version)
         if variant:
-            variant_clause = "products.variant = %(variant)s"
-            qargs['variant'] = variant
-        else:
-            variant_clause = "products.variant is NULL"
-
-        qry = """
-            SELECT name, pkg_arch, product_arch, include
-            FROM overrides
-            WHERE product IN (
-                SELECT id
-                FROM products
-                WHERE label = %(product)s
-                AND version = %(version)s
-                AND """ + variant_clause + """
-            )
-            """
-
-        dbc = compose_dbh.cursor()
-        Products.execute_query(dbc, qry, **qargs)
-        rows = dbc.fetchall()
+            query = query.filter(models.Products.variant == variant)
 
         overrides = {}
-        for row in rows:
-            name, pkg_arch, product_arch, include = row[0:4]
+        for row in query.all():
+            name, pkg_arch, product_arch, include = row.name, row.pkg_arch, row.product_arch, row.include
             overrides.setdefault(name, {}).setdefault(pkg_arch, {}).setdefault(product_arch, include)
         return overrides
     get_overrides = staticmethod(get_overrides)
 
-    def get_match_versions(compose_dbh, product):
+    def get_match_versions(product):
         '''Returns the list of packages for this product where we must match the version.'''
-        dbc = compose_dbh.cursor()
-        Products.execute_query(dbc, """
-            SELECT name
-            FROM match_versions
-            WHERE product = %(product)s
-            """, product=product)
-        rows = dbc.fetchall()
-        matches = []
-        for row in rows:
-            matches.append(row[0])
-        return matches
+        return [m.name for m in models.MatchVersions.query.filter_by(product=product).all()]
     get_match_versions = staticmethod(get_match_versions)
 
-    def get_srconly_flag(compose_dbh, product, version):
+    def get_srconly_flag(product, version):
         '''BREW-260 - Returns allow_source_only field for the product and matching version.'''
-        dbc = compose_dbh.cursor()
-        Products.execute_query(dbc, """
-            SELECT allow_source_only
-            FROM products
-            WHERE label = %(product)s
-            AND version = %(version)s
-            """, product=product, version=version)
-        rows = dbc.fetchall()
-        for (allow_source_only,) in rows:
-            if allow_source_only:
-                return True
-        return False
+        if models.Products.query.filter_by(label=product, version=version, allow_source_only=True).count():
```

Comment:
```python
return models.Products.query.filter_by(
    label=product, version=version, allow_source_only=True).exists()
```

Reply:
exists() is better than count() but it does not work simply like this.